client/question/services/code_evaluators/PythonCodeRunnerService.js

```diff
@@ -52,7 +58,19 @@ tie.factory('PythonCodeRunnerService', [
       outputLines.push(line);
     };
 
-    var _runCodeInClient = function(code) {
+    var _runCodeInClient = function(inputCode) {
+      var code;
+      var separator;
+
+      // The input code is a PreprocessedCode object and ready to be run.
+      if (typeof (inputCode) === 'object') {
```

Comment:
This seems indirect. I think this method should still do what it used to do, which is to just take a piece of code, run it, and return the output.

This method could then be called by a couple of more "public-facing" methods along the lines of compileCode() and runPreprocessedCode() (or similar). Those methods would each do the processing required to get the code that actually needs to be run, send that code to this helper method, then get the output, parse it, and return it.